node/router/src/handshake.rs

```diff
@@ -301,7 +301,7 @@ impl<N: Network> Router<N> {
             bail!("Dropping connection request from '{peer_ip}' (restricted)")
         }
         // Ensure the peer is not spamming connection attempts.
-        if !peer_ip.ip().is_loopback() {
+        if !self.tcp.is_self_connect(peer_ip) {
```

Comment:
why was this check changed? we want to avoid the connection spam safeguards to not trigger in local tests, not to special-case self-connection attempts